src/components/ListHeader/styles.module.css

```diff
@@ -1,32 +1,132 @@
 .listHeader {
-    background-color: var(--main_green);
-    box-sizing: border-box;
-    padding: 26px 26px 60px 26px;
-    height: 165px;
-    display: flex;
-    flex-direction: column;
-    justify-content: space-between;
+  background-color: var(--main_green);
+  box-sizing: border-box;
+  padding: 26px 26px 60px 26px;
+  height: 100%;
+  display: flex;
+  flex-direction: column;
+  justify-content: space-between;
+  transition: all 1s;
 }
 
-.headerRow {
-    display: flex;
-    justify-content: space-between;
-    align-items: center;
+.headerRow,
+.subheadingRow {
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
+  margin: 15px 0;
 }
 
 .headerRow h3,
-.headerRow h4 {
-    margin: 0;
-    color: #FFFFFF;
-    opacity: 0.7;
-    font-weight: 600;
+.headerRow h4,
+.subheadingRow h4 {
+  margin: 0;
+  color: #FFFFFF;
+  opacity: 0.7;
+  font-weight: 600;
+}
+
+.headerRow h3 {
+  display: flex;
+  align-items: center;
 }
 
 .logo {
-    height: 20px;
+  height: 20px;
 }
 
 .shareIcon {
-    font-size: 20px;
-    color: #FFFFFF;
+  font-size: 20px;
+  color: #FFFFFF;
+  cursor: pointer;
+}
+
+.pencilIcon {
+  margin-left: 15px;
+  font-size: 16px;
+  outline: none;
+  border: 0;
+  cursor: pointer;
+}
+
+.shareIcon:focus,
+.pencilIcon:focus {
+  outline: none;
+  border: 0;
+}
+
+.clockIcon,
+.deleteListIcon {
+  margin-right: 10px;
+}
+
+.subheadingRow {
+  justify-content: center;
+}
+
+.displayListDetails {
+  height: 100%;
+}
+
+.hideListDetails {
+  height: 0;
+  opacity: 0;
```

Comment:
If this is going to work you'll also want to add an `overflow: hidden` here, otherwise the items are still there and are still clickable, they just aren't visible, which is why we get this weird behaviour:
![trimmed mp4](https://user-images.githubusercontent.com/9036408/101071534-a71c1700-3594-11eb-97d6-c38bd72da5d8.gif)